include/openssl/sshkdf.h

```diff
@@ -0,0 +1,42 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENSSL_HEADER_SSHKDF_H
+#define OPENSSL_HEADER_SSHKDF_H
+
+#include <openssl/base.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+
+// SSH-specific KDF
+//
+// This KDF should only be called from SSH client/server code; it's not a
+// general-purpose KDF and is only Approved for FIPS 140-3 use specifically
+// in SSH.
+
+// |type| values for SSHKDF().
+#define EVP_KDF_SSHKDF_TYPE_INITIAL_IV_CLI_TO_SRV     65
+#define EVP_KDF_SSHKDF_TYPE_INITIAL_IV_SRV_TO_CLI     66
+#define EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_CLI_TO_SRV 67
+#define EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_SRV_TO_CLI 68
+#define EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_CLI_TO_SRV  69
+#define EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_SRV_TO_CLI  70
+
+// SSHKDF calculates |out_len| bytes of the SSH KDF, using |digest|, and
+// writes them to |out|. It returns one on success and zero on error.
```

Comment:
I think this is too much detail (the only people who should be using this are just calling this instead of the one built in to OpenSSH), but I've gone with something similar.